### PR TITLE
vscode-extensions.mikestead.dotenv: Add dotenv extension

### PR DESCRIPTION
**Pull request by andyrichardson**
###### Motivation for this change

Add dotenv extension

> Disclaimer: I won't be keeping this up to date as future releases are made.
>
> It looks like there are already automation steps being made to keep extensions up to date.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please change the commit message and title to ``vscode-extensions.mikestead.dotenv: Init at 1.0.1``

File: pkgs/misc/vscode-extensions/default.nix [hunk 1]
```diff
@@ -155,6 +155,18 @@ let
         };
       };
 
+      mikestead.dotenv = pkgs.vscode-utils.buildVscodeMarketplaceExtension {
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/default.nix:158]:
```suggestion
      mikestead.dotenv = buildVscodeMarketplaceExtension {
```

**Comment by andyrichardson** [comment]
> Please change the commit message

Oh jesus not again 😅 

Okay I'll see what I can do - I've got [a bunch of these](https://github.com/andyrichardson/dotfiles/pull/2/files#diff-33728d24bddf4b4ba5b0f220c65e2b6623dbbc51a5f54a664b666133fe7360d9R31-R184) I'd like to contribute (most with at least 1m downloads).

What is the criteria for a valid commit message? No chance of squashing and setting a preferred message on merge?

**Comment by SuperSandro2000** [comment]
> Oh jesus not again 😅

It is required for ofborg to work correctly.

> Okay I'll see what I can do - I've got a bunch of these I'd like to contribute (most with at least 1m downloads).

It would be fine for me if they would be part of one big PR.

> What is the criteria for a valid commit message?

packageName: message

``python3Packages.python-gitlab: Fixed X``

also see https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Comment by andyrichardson** [comment]
> It would be fine for me if they would be part of one big PR.

@SuperSandro2000 thanks for the response! I'll get a PR in with all the packages.

What do you want the name of the commit and the name of the PR to be?

**Comment by SuperSandro2000** [comment]
> What do you want the name of the commit and the name of the PR to be (given there is more than one package involved)?

Please do one commit per extensions and the name can be something like: ``vscode-extensions: Add X,Y,X`` and if this gets to long you can do ``vscode-extensions: Add several extensions``.

**Pull request closed without merging**